be/src/vec/functions/function_struct_in.h

```diff
@@ -0,0 +1,164 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+// This file is copied from
+
+#pragma once
+
+#include <glog/logging.h>
+#include <stddef.h>
+
+#include <algorithm>
+#include <memory>
+#include <unordered_set>
+#include <utility>
+#include <vector>
+
+#include "common/status.h"
+#include "vec/columns/column.h"
+#include "vec/columns/column_const.h"
+#include "vec/columns/column_nullable.h"
+#include "vec/columns/column_struct.h"
+#include "vec/columns/column_vector.h"
+#include "vec/columns/columns_number.h"
+#include "vec/core/block.h"
+#include "vec/data_types/data_type_factory.hpp"
+#include "vec/data_types/data_type_nullable.h"
+#include "vec/data_types/data_type_number.h"
+#include "vec/functions/function.h"
+
+namespace doris::vectorized {
+struct ColumnRowRef {
+    ColumnPtr column;
+    size_t row_idx;
+
+    // equals when call set insert, this operator will be used
+    bool operator==(const ColumnRowRef& other) const {
+        return column->compare_at(row_idx, other.row_idx, *column, 0) == 0;
+    }
+    // compare
+    bool operator<(const ColumnRowRef& other) const {
+        return column->compare_at(row_idx, other.row_idx, *column, 0) < 0;
+    }
+
+    // when call set find, will use hash to find
+    size_t operator()(const ColumnRowRef& a) const {
+        std::vector<uint32_t> hash_vec(1);
```

Comment:
why use vector，not `uint32_t`

Reply:
I tried use uint32_t , but I make some tests , but got different val in hash_val